sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -73,6 +74,34 @@ public BinaryData(string data)
             Bytes = s_encoding.GetBytes(data);
         }
 
+        /// <summary>
+        /// Creates a binary data instance from an object and serializes it
+        /// using the <see cref="JsonObjectSerializer"/>.
+        /// </summary>
+        /// <param name="data">The data that will be serialized.</param>
+        /// <param name="type">The type of the data.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public BinaryData(object data, Type type) :
+            this (data, type, new JsonObjectSerializer())
+        {
+        }
+
+        /// <summary>
+        /// Creates a binary data instance from an object and serializes it
+        /// using the provided <see cref="ObjectSerializer"/>.
+        /// </summary>
+        /// <param name="data">The data that will be serialized.</param>
+        /// <param name="type">The type of the data.</param>
+        /// <param name="serializer">The serializer to serialize
+        /// the data.</param>
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        public BinaryData(object data, Type type, ObjectSerializer serializer)
```

Comment:
What about 
>This should not expose ObjectSerializer parameter as that would mean we would need to support async, which we can't do in the constructor. Using this constructor would use JsonObjectSerializer.
?

Reply:
I think this is a reasonable compromise - using ObjectSerializer requires the static methods.

Reply:
Default to null - use JsonObjectSerializer.
Remove other ctor.